## Introduction

Panmask provides a list of easy/hard regions for short-read variant calling against the human genome GRCh38.
The easy regions harbor small variants that are easy to call, with most variant callers achieving 98-99.5% accuracy in the regions.
They cover 87.9% of GRCh38, 92.6% of coding regions and 95.8% of pathogenic variants in ClinVar.
The panmask regions may help to reduce variant calling artifacts and simplify variant filtering.
They can be downloaded [from Zenodo][zenodo].

[zenodo]: https://zenodo.org/records/14903542

## Other Datasets

GRCh38 easy regions (where variant calls tend to be accurate in most samples):

* [umap-k100][umap]: Umap for 100bp single-end reads, published in [Karimzadeh et al (2018)][pub-umap]
* [ENCODE][encode]: ENCODE blacklist regions v2, published in [Amemiya et al (2019)][pub-encode]
* [GIAB-easy][GIAB-easy]: GIAB genome stratification v3.5, published in [Dwarshuis et al (2024)][pub-GIAB-easy]
* [Illumina][il-t2t]: originally developed [by Illumina][il-ori], reimplemented by Taylor and McCoy from JHU.
* [1000G mask][1kg]: developed for the 1000 Genomes Project. Developers unknown.
* [highRepro][highrep]: intersection of highly reproducible regions generated by [Pan et al (2022)][pub-highRepro]

[GIAB-easy]: https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/genome-stratifications/v3.5/GRCh38@all/Union/
[pub-GIAB-easy]: https://www.nature.com/articles/s41467-024-53260-y
[umap]: https://bismap.hoffmanlab.org
[pub-umap]: https://academic.oup.com/nar/article/46/20/e120/5086676?login=false
[encode]: https://github.com/Boyle-Lab/Blacklist
[pub-encode]: https://www.nature.com/articles/s41598-019-45839-z
[pub-highRepro]: https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02569-8
[il-t2t]: https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=T2T/CHM13/assemblies/annotation/accessibility/
[il-ori]: https://www.illumina.com/science/genomics-research/articles/identifying-genomic-regions-with-high-quality-single-nucleotide-.html
[1kg]: https://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/working/20160622_genome_mask_GRCh38/
[highrep]: https://hgdownload.soe.ucsc.edu/gbdb/hg38/problematic/highRepro/

HG002 confident regions (where small variant calls can be trusted):

* [HG002-GIAB][GIAB-old]: NIST confident regions, v4.2.1
* [HG002-Q100][GIAB-new]: T2TQ100 v1.1/20241113

[GIAB-old]: https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/AshkenazimTrio/HG002_NA24385_son/NISTv4.2.1/GRCh38/
[GIAB-new]: https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/NIST_HG002_DraftBenchmark_defrabbV0.019-20241113/

Other datasets used for evaluation:

* ClinVar, [20250623][clinvar-dw]
* GenCode basic gene annotation, [v48][gencode-gtp]
* gnomAD, [v4.1.0][gnomad]

[clinvar-dw]: https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/archive_2.0/2025/
[gencode-gtp]: https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_48/

Short-read [small variant calls][varcall], published in [Baid et al (2020)][pub-varcall].
Only VCFs called from HG002 PCR-free NovaSeq data at 30X are used.
Data files in this repo are released under CC0 and will be [available at GigaDB][gigadb].

[varcall]: https://console.cloud.google.com/storage/browser/brain-genomics-public/research/sequencing/grch38/vcf/novaseq/wgs_pcr_free/30x
[pub-varcall]: https://www.biorxiv.org/content/10.1101/2020.12.11.422022v1
[gigadb]: https://doi.org/10.5524/102750
[gnomad]: https://gnomad.broadinstitute.org
