|                    |GIAB-easy| ENCODE  |umap-k100| pm151   | pm151a  |HG002-Q100|HG002-GIAB|
|:-------------------|--------:|--------:|--------:|--------:|--------:|---------:|---------:|
|Size (Gb)           |2.31     |2.83     |2.82     |2.55     |2.59     |2.74      |2.54      |
|# blocks            |4.84M    |731      |242k     |1.20M    |299k     |28.7k     |481k      |
|N50 block length    |860      |16.1M    |142k     |4158     |22.9k    |183k      |11.2k     |
|% GRCh38            |79.0     |96.8     |96.7     |87.2     |88.7     |93.7      |87.0      |
|% GenCode CDS       |72.6     |98.5     |97.6     |92.0     |92.3     |95.8      |91.8      |
|% ClinVar patho     |77.4     |99.8     |99.6     |95.1     |95.8     |98.8      |90.2      |
|SNP %FNR (DV/GATK)  |0.14/0.15|3.34/3.40|4.81/4.84|0.38/0.42|1.00/0.98|1.28/1.21 |0.66/0.41 | 
|SNP %FDR (DV/GATK)  |0.32/0.85|1.37/3.81|2.61/6.69|0.40/1.43|0.53/1.56|0.22/1.33 |0.20/1.15 |
|SNP FPpM (DV/GATK)  |4.1/10.7 |18.3/52.0|35.4/94.3|5.1/18.4 |6.8/20.3 |3.0/17.7  |2.7/15.3  |
|INDEL %FNR (DV/GATK)|2.79/2.50|7.62/7.95|8.07/8.37|2.49/2.14|5.18/5.42|3.22/3.94 |0.43/0.41 |
|INDEL %FDR (DV/GATK)|0.92/2.42|2.01/3.20|2.51/4.44|0.74/1.98|1.37/2.09|0.76/1.22 |0.14/0.59 |
|INDEL FPpM (DV/GATK)|0.8/2.0  |7.1/11.4 |8.9/16.1 |1.4/3.8  |4.8/7.3  |2.6/4.2   |0.3/1.3   |

<!--
|% gnomAD SNP20 pass| 89.8        | 78.2    | 99.6       |
|% gnomAD SNP20 flt | 26.8        | 7.1     | 86.5       |

* % gnomAD SNP20 pass: percent PASS SNPs at 20% allele frequency in gnomAD are included in the confident regions
* % gnomAD SNP20 flt: percent filtered SNPs at 20% allele frequency in gnomAD are included in the confident regions
* ClinVar version: 20250209
* GenCode version: v47
* Q100 version: 20241113
-->
